eftest/src/eftest/runner.clj

```diff
@@ -16,24 +16,46 @@
   (or (-> v meta :eftest/synchronized true?)
       (-> v meta :ns meta :eftest/synchronized true?)))
 
+(defn- known-slow? [v]
+  (or (-> v meta :eftest/slow true?)
+      (-> v meta :ns meta :eftest/slow true?)))
+
 (defn- failed-test? []
   (or (< 0 (:error @test/*report-counters* 0))
       (< 0 (:fail @test/*report-counters* 0))))
 
+(defn- wrap-test-with-timer [test-warn-time v test-fn]
+  (let [start-time (System/nanoTime)
+        result   (test-fn)
+        end-time (System/nanoTime)
+        duration (/ (- end-time start-time) 1e6)]
+    (when (and (not (known-slow? v))
+               (number? test-warn-time)
+               (<= test-warn-time duration))
+      (binding [clojure.test/*testing-vars* (conj clojure.test/*testing-vars* v)]
+        (test/report {:type     :long-test
+                      :duration duration
+                      :var      v})))
+    result))
+
 (defn- test-vars
-  [ns vars {:as opts :keys [fail-fast? capture-output?] :or {capture-output? true}}]
+  [ns vars {:as opts :keys [fail-fast? capture-output? test-warn-time]
+            :or {capture-output? true}}]
   (let [once-fixtures (-> ns meta ::test/once-fixtures test/join-fixtures)
         each-fixtures (-> ns meta ::test/each-fixtures test/join-fixtures)
         report        (synchronize test/report)
         test-var      (fn [v]
                         (when-not (and fail-fast? (failed-test?))
-                          (each-fixtures
-                           (if capture-output?
-                             #(binding [test/report report]
-                                (capture/with-test-buffer
-                                  (test/test-var v)))
-                             #(binding [test/report report]
-                                (test/test-var v))))))]
+                          (wrap-test-with-timer test-warn-time
```

Comment:
I think it would look cleaner to put the wrapper on line 61 instead, e.g.:

```clojure
(let [test (bound-fn* (wrap-test-with-timer test-var v test-warn-time)]
```

Reply:
done